### PR TITLE
17671 - add futureEffective for alteration

### PR DESCRIPTION
**Pull request by tshyun24**
*Issue #:* /bcgov/entity#17671

*Description of changes:*
Add futureEffective when doing alteration

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the lear license (Apache 2.0).

### CODE REVIEW

**Comment by argush3** [comment]
As per our conversation during standup, we will fix this issue in a minimal effort way and re-factor the filing type fee code retrieval function in another tech debt ticket(#18479).  cc: @seeker25

**Comment by argush3** [comment]
Can we update the [test_get_corrrect_fee_codes](https://github.com/bcgov/lear/blob/c2e4af61374e77f489708d0da67e3ba4692c8f79/legal-api/tests/unit/resources/v2/test_business_filings/test_filings.py#L1113) tests to include checks for future effective date scenarios? 

I was thinking maybe we maybe add a `has_fed` flag to the parameterized test and use it to test that IA, voluntary dissolution and alteration filings properly includes the `futureEffective` property/value when [get_filing_types](https://github.com/bcgov/lear/blob/2f619e9d2cd073b746bbcea53a93310bcdf50efc/legal-api/src/legal_api/resources/v2/business/business_filings/business_filings.py#L637) is called.

**Comment by seeker25** [comment]
Another thing we probably need Argus - the filing fee codes should come from the same source of truth. The frontend has it's own version with the enums in BCRS-SHARED-COMPONENTS, where as the backend reads from filings.py in models. 
This can cause some inconsistencies in the future. Maybe we could serve filings.py from an endpoint and have the frontend just call into LEAR to get the fee codes?

File: legal-api/tests/unit/resources/v2/test_business_filings/test_filings.py [hunk 1]
```diff
@@ -46,7 +46,7 @@
 )
 
 from legal_api.models import Business, Filing, RegistrationBootstrap, UserRoles
-from legal_api.resources.v1.business.business_filings import ListFilingResource
+from legal_api.resources.v2.business.business_filings.business_filings import ListFilingResource
```
Comment (argush3) [inline legal-api/tests/unit/resources/v2/test_business_filings/test_filings.py:49]: thanks for catching this!

**Review by argush3 — APPROVED** [review]

**Review by chenhongjing — APPROVED** [review]
Looks good 😺

**Pull request merged**